### ISSUE TITLE
rename all doesn't always work (at all)

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap--Build-Your-Own-Blocks#2085]
I assume this is because I've used it in a dropdown, but I don't know. 

![rename-all](https://user-images.githubusercontent.com/6549787/39076341-ba737b5a-44c8-11e8-9cdd-c063c2802e75.gif)

This particular instance of snap has been buggy all day. Earlier it cleaned-up within a block editor by putting things above the hat block and then they were all crammed on top of each other when I reopened it. This instance of snap has been open for about 4.5 hours.

**Comment by maryfries** [comment]
I'm finding that sometimes rename all works, sometimes it will change only the variable in the hat box and not close the dialog box, and sometimes it doesn't to anything at all.

**Comment by ToonTalk** [comment]
This happened to me just now and the console shows this exception

blocks.js?version=2018-03-09:3332 Uncaught TypeError: script.refactorVarInStack is not a function
    at blocks.js?version=2018-03-09:3332
    at Array.forEach (<anonymous>)
    at ReporterBlockMorph.BlockMorph.doRefactorBlockParameter (blocks.js?version=2018-03-09:3331)
    at ReporterBlockMorph.renameVarTo (blocks.js?version=2018-03-09:3280)
    at DialogBoxMorph.accept (widgets.js?version=2018-03-09:2310)
    at DialogBoxMorph.ok (widgets.js?version=2018-03-09:2352)
    at PushButtonMorph.TriggerMorph.trigger (morphic.js?version=2018-03-09:9741)
    at PushButtonMorph.TriggerMorph.mouseClickLeft (morphic.js?version=2018-03-09:9802)
    at PushButtonMorph.mouseClickLeft (widgets.js?version=2018-03-09:209)
    at StringMorph.Morph.escalateEvent (morphic.js?version=2018-03-09:4351)

**Comment by maryfries** [comment]
Still having this problem. It works for input variables but not for script variables.

**Comment by jguille2** [comment]
Yes Mary,
I can reproduce your issue. The "rename all" for script variables is not working inside the block editor.

While this problem is not fixed... a workaround is to move the whole script to the "script area" (out of the block editor), "rename all" there (is working fine) and move again to the "block editor".

Joan

**Comment by maryfries** [comment]
Thanks, but it's the input variables that I've been trying to change. Maybe if I pull a script out and then put a script variables block on there with variables of the same names and then changed them, it would work. I can try that.

There doesn't seem to be any way to rename all for globals either.

Mostly I've been getting it to work sometimes, and doing the rest by hand. I haven't found any pattern to when it works yet, but I'll post something if I do. At this point, it seems to work for some variables and not others—even within the same block definition and regardless of input type. Reopening the project doesn't seem to impact which work and which don't.

**Comment by maryfries** [comment]
Ah yes, I've run into this issue in different ways over time, it seems. The issue of renaming input variables is unpredictable; sometimes it works, and sometimes it doesn't. Brian was there with me and can confirm. ;) Thanks!

**Comment by jmoenig** [comment]
Yeah, refactoring variables seems unnecessarily complicated in the source code, I believe you that it sometimes doesn't work. It's only my list to revisit this when I get around to it. Bear with me!
